aicsimageio/aics_image.py

```diff
@@ -191,8 +191,14 @@ def data(self) -> np.ndarray:
         """
         Return the entire image as a numpy array with dimension ordering "STCZYX".
         """
-        if self._data is None:
+        if self._data is None and self.reader._data is None:
             self._data = self.dask_data.compute()
+        else:
+            self._data = transforms.reshape_data(
+                data=self.reader.data,
+                given_dims=self._known_dims or self.reader.dims,
+                return_dims=self.dims,
+            )
```

Comment:
is this still an optimization or is it just extraneous now, since reshape_data would be called anyway

Reply:
Now that everything happens on `.data` this is still needed but we don't need the second `if` check